agent/src/com/thoughtworks/go/agent/service/SslInfrastructureService.java

```diff
@@ -148,25 +149,28 @@ private void deleteKeyStores() {
 
     public static class RemoteRegistrationRequester {
         private final AgentRegistry agentRegistry;
+        private SystemEnvironment systemEnvironment;
         private String serverUrl;
         private GoAgentServerHttpClient httpClient;
 
-        public RemoteRegistrationRequester(String serverUrl, AgentRegistry agentRegistry, GoAgentServerHttpClient httpClient) {
+        public RemoteRegistrationRequester(String serverUrl, AgentRegistry agentRegistry, GoAgentServerHttpClient httpClient, SystemEnvironment systemEnvironment) {
             this.serverUrl = serverUrl;
             this.httpClient = httpClient;
             this.agentRegistry = agentRegistry;
+            this.systemEnvironment = systemEnvironment;
         }
 
         protected Registration requestRegistration(String agentHostName, AgentAutoRegistrationProperties agentAutoRegisterProperties) throws IOException, ClassNotFoundException {
             if (LOGGER.isDebugEnabled()) {
                 LOGGER.debug(String.format("[Agent Registration] Using URL %s to register.", serverUrl));
             }
 
+            String agentWorkDir = systemEnvironment.resolveAgentWorkingDirectory(new File(".")).getAbsolutePath();
             HttpRequestBase postMethod = (HttpRequestBase) RequestBuilder.post(serverUrl)
                     .addParameter("hostname", agentHostName)
                     .addParameter("uuid", agentRegistry.uuid())
-                    .addParameter("location", SystemUtil.currentWorkingDirectory())
```

Comment:
I am trying ... and there are 150 usages in tests. I could get it down to 20 usages by extracting a method in each of those tests and then I'd need to inline it. Do you think it's worth it, for what is essentially `new File(".").getAbsolutePath`?